src/ucp/api/ucp.h

```diff
@@ -591,6 +591,13 @@ enum ucp_send_am_flags {
     UCP_AM_SEND_FLAG_RNDV  = UCS_BIT(2),             /**< Force UCP to use only
                                                           rendezvous protocol for
                                                           AM sends. */
+    UCP_AM_SEND_FLAG_COPY_HEADER  = UCS_BIT(3),     /**< Valid user header during send 
+                                                         operation is not guaranteed, 
+                                                         Signal UCP to copy it when needed*/
+    UCP_AM_SEND_PRIV_FLAG_FREE_HEADER  = UCS_BIT(4),     /**< Internal UCP Flag, 
```

Comment:
Use another enum for `PRIV` flags (see `UCP_AM_CB_PRIV_FIRST_FLAG` for example) - these should not be exposed to the user.